releasenotes/notes/fix-instruction-repeat-conditional-dfe4d7ced54a7bb6.yaml

```diff
@@ -0,0 +1,7 @@
+---
+fixes:
+  - |
+    :meth:`.Instruction.repeat` will now correctly move a set :attr:`~.Instruction.condition` to the
+    outer returned :class:`~.circuit.Instruction` and leave the inner gates of its definition
+    unconditional.  Previously, the method would leave :class:`.ClassicalRegister` instances within
+    the inner definition, which broke the data model.  Fixed `#11935 <https://github.com/Qiskit/qiskit/issues/11935>`__.
```

Comment:
The explanation of the bug that is fixed can be more user oriented, I think. "broke the data model" is technically correct, but the user see the manifestation of that while transpiling or qasm3-dumping dynamic circuits with `repeat`.

Reply:
Sure, done in b8361fb14a8.